scheduler/src/main/java/io/camunda/zeebe/scheduler/future/CompletableActorFuture.java

```diff
@@ -210,8 +216,23 @@ public void block(final ActorTask onCompletion) {
 
   @Override
   public void onComplete(final BiConsumer<V, Throwable> consumer) {
-    final ActorControl actorControl = ActorControl.current();
-    actorControl.runOnCompletion(this, consumer);
+    if (ActorThread.isCalledFromActorThread()) {
+      final ActorControl actorControl = ActorControl.current();
+      actorControl.runOnCompletion(this, consumer);
+    } else {
+      // We don't reject this because, this is useful for tests. But the warning is a reminder not
+      // to use this in production code.
```

Comment:
:wrench: I suppose you already had use cases in mind but I'm a bit sad that this ends up in the public API. Couldn't we require tests to either:
1. Ad-hoc use of `Executors.newSingleThreadExecutor()` (or similar executors)
2. Use a helper class that manages a small executor that can be used by tests

Reply:
Let say we have a method as below:
```
void productionMethod() {
{
   ..
   future.onComplete(this::handler)
}
```
In production code, we will always invoke this method from an actor. But if we want to test this method, we don't want to use actor. In some cases, we can control the type of the `future` and make it `TestActorFuture`. But it is not always possible or will result in more complex test code.. To allow testing, we want to allow the same code to work both on actor and non-actor. This is why I want to allow it to be called in non-actor context. Alternative is always provide an executor, which is also not nice and might change the current guarantees of when tasks are run after a future is completed.